source/OptimaJet.DWKit.StarterApplication/Repositories/UserRepository.cs

```diff
@@ -26,46 +26,47 @@ IDbContextResolver contextResolver
         {
-            if (OrganizationContext.SpecifiedOrganizationDoesNotExist) return Enumerable.Empty<User>().AsQueryable();
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist) {
+                return Enumerable.Empty<User>().AsQueryable();
+            }
+
+            var currentUser = GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
+
+            var query = base.Get();
+
             if (!OrganizationContext.HasOrganization)
             {
-                // No organization specified, so include all users in the all the organizations that the current user is a member
-                var currentUser = GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
-                return DefaultGet()
-                           .Where(u => u.OrganizationMemberships.Select(o => o.OrganizationId).Intersect(currentUser.OrganizationIds).Any());
-            }
-            // Get users in the current organization
-            return DefaultGet()
-                       .Where(u => u.OrganizationMemberships.Any(o => o.OrganizationId == OrganizationContext.OrganizationId));
-        }
+                // No organization specified, so include all users in the all the 
+                // organizations that the current user is a member
+                var orgIds = currentUser.OrganizationIds ?? Enumerable.Empty<int>();
 
-        public override async Task<User> GetAsync(int id)
-        {
-            // The default implementation filters by selected organization.  The
-            // current user might not be in that organization.
-            // Always allow getting the current user.
-            var currentUser = await GetByAuth0Id(CurrentUserContext.Auth0Id);
-            if (currentUser != null && currentUser.Id == id) {
-                return await DefaultGet().Where(e => e.Id == id).FirstAsync();
+                return query
+                    .Where(u => (
+                        u.Id == currentUser.Id 
+                        || (u.OrganizationMemberships
+                                .Select(o => o.OrganizationId)
+                                .Intersect(orgIds)
+                                .Any()
+                            )
+                    ));
             }
-            return await base.GetAsync(id);
+
+            // Get users in the current organization
+            return query
+                .Where(u => (
+                    u.Id == currentUser.Id 
+                    || (u.OrganizationMemberships
+                            .Any(o => o.OrganizationId == OrganizationContext.OrganizationId)
+                        )
+                ));
         }
 
         public async Task<User> GetByAuth0Id(string auth0Id)
             => await base.Get()
                 .Where(e => e.ExternalId == auth0Id)
-                .Include(u => u.OrganizationMemberships)
-                    .ThenInclude(om => om.Organization)
+                .Include(user => user.OrganizationMemberships)
```

Comment:
only the OrganizationMembership is included here, because it is used above in the `Intersect`. It's also how I learned about the possible 500 exception when there are no OrganizationMemberships (which was the behavior when this Include was omitted).

The Organization is not needed anywhere as far as I could tell. (at least no tests create a situation where it needs to be loaded)